### PR TITLE
squash GCC 12.1.1 complaints

### PR DESCRIPTION
**Pull request by ksmurchison**
Just upgraded OS on my laptop which came with newer GCC

Should be fix for #4085

### CODE REVIEW

File: imap/ical_support.c [hunk 1]
```diff
@@ -2368,7 +2368,7 @@ EXPORTED void icaltimezone_truncate_vtimezone_advanced(icalcomponent *vtz,
     icaltimetype start = *startp, end = *endp;
     icalcomponent *comp, *nextc, *tomb_std = NULL, *tomb_day = NULL;
     icalproperty *prop, *proleptic_prop = NULL;
-    struct observance tombstone;
+    static struct observance tombstone;
```
Comment (rsto) [inline imap/ical_support.c:2371]: Could we instead require callers to pass a `struct observance*` instead?

**Review by rsto — CHANGES_REQUESTED** [review]

File: imap/ical_support.c (hunk 1 above)
Comment (ksmurchison) [reply imap/ical_support.c]: Fixed

**Review by rsto — APPROVED** [review]
LGTM

File: imap/ical_support.c [hunk 2]
```diff
@@ -2847,7 +2847,9 @@ EXPORTED void icaltimezone_truncate_vtimezone_advanced(icalcomponent *vtz,
         }
     }
 
-    if (proleptic) *proleptic = &tombstone;
+    if (proleptic) {
+        memcpy(proleptic, &tombstone, sizeof(struct observance));
+    }
```
Comment (dilyanpalauzov) [inline imap/ical_support.c:2852]: Will `if (proleptic) *proleptic = tombstone` work?

**Pull request merged**